tools/src/scie_pants/pants_version.py

```diff
@@ -117,12 +120,22 @@ def determine_tag_version(
         github_api_url = (
             f"https://api.github.com/repos/pantsbuild/pants/git/refs/tags/{urllib.parse.quote(tag)}"
         )
+        github_releases_url="https://github.com/pantsbuild/pants/releases"
         headers = (
             {"Authorization": f"Bearer {github_api_bearer_token}"}
             if github_api_bearer_token
             else {}
         )
-        github_api_tag_url = ptex.fetch_json(github_api_url, **headers)["object"]["url"]
+        github_api_response: Dict[str, Any]
+        try:
+            github_api_response = ptex.fetch_json(github_api_url, **headers)
+            # If the response is a list, multiple matches were found, which means the supplied tag
+            # is not an exact match against any tag
+            if isinstance(github_api_response, list):
```

Comment:
It seems like it would make a but more sense to keep the comment mostly unchanged but change the test to `if not isinstance(github_api_response, dict)` since that is what is used / relied upon on line 138 below. As it stands, if the API returned a string a bool or a number, the same style of confusing exception as this fixes would be raised.